File: Python/2545. Sort the Students by Their Kth Score.py
class Solution:
    def sortTheStudents(self, score: List[List[int]], k: int) -> List[List[int]]:
        temp = {x: y for x, y in enumerate([score[x][k] for x in range(len(score))])}
        temp = sorted(temp.items(), key=lambda x: x[1], reverse=True)
        temp = [x[0] for x in temp]

        res = [[] for _ in range(len(score))]

        for i, x in enumerate(temp): res[i] = score[x]

        return res

File: C++/2545. Sort the Students by Their Kth Score.cpp
class Solution {
public:
    vector<vector<int>> sortTheStudents(vector<vector<int>>& score, int k) {
        vector<int> temp;
        map<int, int> dict;
        vector<vector<int>> res = score;

        for (int i = 0; i < score.size(); i++) {
            temp.push_back(score[i][k]);
            dict[score[i][k]] = i;
        }

        sort(temp.begin(), temp.end());

        for (int i = 0; i < temp.size(); i++) res[temp.size() - i - 1] = score[dict[temp[i]]];

        return res;
    }
};
